awx/main/analytics/certificate_manager.py

```
@@ -0,0 +1,439 @@
+    def get_serial_number(self):
+        """
+        Get the serial number from the local certificate.
+
+        Adapted from subscription_manager/identity.py ConsumerIdentity.
+        """
+        try:
+            from cryptography import x509
+
+            with open(self.certpath, 'rb') as f:
+                cert = x509.load_pem_x509_certificate(f.read())
+            return cert.serial_number
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)